pkg/operatorsource/handler.go

```diff
@@ -16,36 +25,31 @@ type Handler interface {
 }
 
 type handler struct {
-	reconciler Reconciler
+	strategy     PhaseReconcilerStrategy
+	kubeclient   Kubeclient
+	transitioner PhaseTransitioner
 }
 
 func (h *handler) Handle(ctx context.Context, event sdk.Event) error {
-	opsrc := event.Object.(*v1alpha1.OperatorSource)
-
-	if event.Deleted {
-		logrus.Infof("No action taken, object has been deleted [type=%s object=%s/%s]",
-			opsrc.TypeMeta.Kind, opsrc.ObjectMeta.Namespace, opsrc.ObjectMeta.Name)
-
-		return nil
+	in, ok := event.Object.(*v1alpha1.OperatorSource)
```

Comment:
Please `err` here too unless of course `Error` is not being return on bad cast. In general it is good practice to limit the number of variables we are creating.

Reply:
`ok` is `bool` type - per type assertion specification.